this.context = Script.createContext();                         ^ TypeError: undefined is not a function

```
myuser-netbook$npm run swank

> swank-js@0.0.5 swank /home/myuser/programs/swank-js-master
> node ./swank.js

/home/myuser/programs/swank-js-master/swank-handler.js:303
  this.context = Script.createContext();
                        ^
TypeError: undefined is not a function
    at new DefaultRemote (/home/myuser/programs/swank-js-master/swank-handler.js:303:25)
    at new Executive (/home/myuser/programs/swank-js-master/swank-handler.js:370:21)
    at Object.<anonymous> (/home/myuser/programs/swank-js-master/swank.js:43:17)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3

npm ERR! Linux 3.2.0-4-686-pae
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "swank"
npm ERR! node v0.12.4
npm ERR! npm  v2.10.1
npm ERR! code ELIFECYCLE
npm ERR! swank-js@0.0.5 swank: `node ./swank.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the swank-js@0.0.5 swank script 'node ./swank.js'.
npm ERR! This is most likely a problem with the swank-js package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./swank.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls swank-js
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/myuser/programs/swank-js-master/npm-debug.log
```

--- a/swank-handler.js
+++ b/swank-handler.js
@@ -300,7 +300,10 @@ Remote.prototype.sendResult = function sendResult (id, values) {
 };
 
 function DefaultRemote () {
-  this.context = Script.createContext();
+    // https://github.com/jashkenas/coffeescript/issues/3498
+    // newer version of node uses vm.createContext(), so Script.createContext fails
+    this.context = (Script.createContext || vm.createContext)(); 
+    
   for (var i in global) this.context[i] = global[i];
   this.context.module = module;
   this.context.require = function(id, options) {

--- a/swank.js
+++ b/swank.js
@@ -272,27 +272,7 @@ HttpListener.prototype.doProxyRequest = function doProxyRequest (targetUrl, requ
   request.headers["host"] = hostname + (port == 80 ? "" : ":" + port);
   delete request.headers["accept-encoding"]; // we don't want gzipped pages, do we?
 
-  // note on http client error handling:
-  // http://rentzsch.tumblr.com/post/664884799/node-js-handling-refused-http-client-connections
-  var proxy = http.createClient(port, hostname);
-  proxy.addListener(
-    'error', function handleError (e) {
-      console.log("proxy error: %s", e);
-      if (done)
-        return;
-      if (headersSent) {
-        response.end();
-        return;
-      }
-      response.writeHead(502, {'Content-Type': 'text/plain; charset=utf-8'});
-      response.end("swank-js: unable to forward the request");
-  });
-
-  console.log("PROXY: %s %s", request.method, request.url);
-  var proxyRequest = proxy.request(request.method, request.url, request.headers);
-
-  proxyRequest.addListener(
-    'response', function (proxyResponse) {
+    var onResponse = function (proxyResponse) {
       var contentType = proxyResponse.headers["content-type"];
       var statusCode = proxyResponse.statusCode;
       console.log("==> status %s", statusCode);
@@ -309,7 +289,7 @@ HttpListener.prototype.doProxyRequest = function doProxyRequest (targetUrl, requ
         response.writeHead(statusCode, headers);
         headersSent = true;
       }
-      proxyResponse.addListener(
+      proxyResponse.on(
         'data', function (chunk) {
           if (chunks !== null) {
             chunks.push(chunk);
@@ -321,7 +301,7 @@ HttpListener.prototype.doProxyRequest = function doProxyRequest (targetUrl, requ
           }
           response.write(chunk, 'binary');
         });
-      proxyResponse.addListener(
+      proxyResponse.on(
         'end', function() {
           if (chunks !== null) {
             console.log("^^MOD: %s %s", request.method, request.url);
@@ -345,12 +325,41 @@ HttpListener.prototype.doProxyRequest = function doProxyRequest (targetUrl, requ
           response.end();
           done = true;
         });
-    });
-  request.addListener(
+    };
+  // note on http client error handling:
+    // http://rentzsch.tumblr.com/post/664884799/node-js-handling-refused-http-client-connections
+    
+    // this is deprecated
+    // var proxy = http.createClient(port, hostname);
+    var proxyRequest = http.request({
+	hostname:hostname, 
+	port:port, 
+	path:request.url,
+	method:request.method,
+	headers:request.headers, 
+    }, onResponse);
+    
+  proxyRequest.on(
+    'error', function handleError (e) {
+      console.log("proxy error: %s", e);
+      if (done)
+        return;
+      if (headersSent) {
+        response.end();
+        return;
+      }
+      response.writeHead(502, {'Content-Type': 'text/plain; charset=utf-8'});
+      response.end("swank-js: unable to forward the request");
+  });
+
+  console.log("PROXY: %s %s", request.method, request.url);
+  // var proxyRequest = proxy.request(request.method, request.url, request.headers);
+  
+  proxyRequest.on(
     'data', function(chunk) {
       proxyRequest.write(chunk, 'binary');
     });
-  request.addListener(
+  proxyRequest.on(
     'end', function() {
       proxyRequest.end();
     });

